src/test/java/io/cdap/delta/datastream/util/MockSourceContext.java

```diff
@@ -28,17 +28,21 @@
 
 import java.io.IOException;
 import java.util.Collections;
+import java.util.HashMap;
 import java.util.Map;
 import java.util.Set;
 import java.util.stream.Collectors;
 
+import static io.cdap.delta.datastream.DatastreamDeltaSource.BUCKET_CREATED_BY_CDF;
+
 public class MockSourceContext implements DeltaSourceContext {
   private String runId;
   private String namespace;
   private String appName;
   private long generation;
   private Set<String> oracleTables;
   private String oracleDb;
+  private Map<String, byte[]> mockState;
```

Comment:
Initialize here otherwise it will be null in empty constructor
```suggestion
  private Map<String, byte[]> mockState = new HashMap<>();
```